### PR TITLE
:sparkles: Lazy loading images

### PR DESCRIPTION
**Pull request by sssuneeth**
Now using custom **Intersection Observer** for this purpose.
And using custom **Image** and **ImageLoader** components for optimizing image.

- [x] Image loads only when comes to viewport
- [x] Shows an animation while loading
- [x] Smooth transition when loaded

### CODE REVIEW

**Comment by baseplate-admin** [comment]
Dont use via placeholder

**Comment by baseplate-admin** [comment]
Also it doesnt work properly

**Comment by sssuneeth** [comment]
> Dont use via placeholder

Okay, here is an alternative [lorem picsum](https://picsum.photos/)
> Also it doesnt work properly

?? which type of placeholder you expect?

**Comment by baseplate-admin** [comment]
> Okay, here is an alternative [lorem picsum](https://picsum.photos/)

Why not ship the placeholder into the frontend app?

> ?? which type of placeholder you expect?

Send a video? it doesn't show any image on firefox android

**Comment by sssuneeth** [comment]
Hey @baseplate-admin 
Check new configurations
>Why not ship the placeholder into the frontend app?

Yea, we can do that in the `else` case here
```html
<IntersectionObserver
    once={true}
    let:intersecting
>
    {#if intersecting}
        <Image
            class={klass}
            {src}
            {alt}
        />
    {:else}
    <!-- we can add another html place holder here with alt name -->
        <span class={klass}>{alt}</span>
    {/if}
</IntersectionObserver>
```

File: package.json [hunk 1]
```diff
@@ -54,6 +54,7 @@
         "svelte": "^3.59.1",
         "svelte-check": "^3.3.2",
         "svelte-gestures": "^1.5.0",
+        "svelte-lazy-image": "^0.5.1",
```
Comment (baseplate-admin) [inline package.json:57]: Remove this

**Review by baseplate-admin — CHANGES_REQUESTED** [review]

File: src/lib/components/shared/image/image_loader.svelte [hunk 2]
```diff
@@ -6,6 +6,7 @@
     export let alt: string;
     let klass: string;
     export { klass as class };
+    export let placeholder_class: string;
```
Comment (baseplate-admin) [inline src/lib/components/shared/image/image_loader.svelte:9]: Why do you need to add this?

**Review by baseplate-admin — CHANGES_REQUESTED** [review]

File: src/lib/components/shared/image/image_loader.svelte [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    export let alt: string;
+    let klass: string;
+    export { klass as class };
+</script>
+
+<IntersectionObserver
+    once={true}
+    let:intersecting
+>
+    {#if intersecting}
```
Comment (baseplate-admin) [inline src/lib/components/shared/image/image_loader.svelte:15]: add a `<div class='grid'>` before ` {#if intersecting} `

File: src/lib/components/shared/image/image_loader.svelte [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        <Image
+            class={klass}
+            {src}
+            {alt}
+        />
```
Comment (baseplate-admin) [inline src/lib/components/shared/image/image_loader.svelte:20]: Add `style='grid-area:1/1/1/1'`

File: src/lib/components/shared/image/image_loader.svelte [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    {:else}
+        <div
+            class="placeholder h-full w-full animate-pulse !bg-surface-400"
+            class:absolute={klass.includes("absolute")}
+        />
```
Comment (baseplate-admin) [inline src/lib/components/shared/image/image_loader.svelte:25]: Add style='grid-area:1/1/1/1'

**Review by baseplate-admin — COMMENTED** [review]
Fix grid like this

**Comment by sssuneeth** [comment]
@baseplate-admin 
Lazy loading will be added on genres desktop view cards later, because we need to change its design a bit.

**Comment by baseplate-admin** [comment]
Sure

**Pull request merged**